test/VoteManager.js

```diff
@@ -638,6 +638,84 @@ describe('VoteManager', function () {
         //
         await assertRevert(tx3, 'sorted[i] is not greater than lastVisited');
       });
+      it('Block should not be proposed when no one votes', async function () {
+        await mineToNextEpoch();
+        const epoch = await getEpoch();
+        // commit state
+        await mineToNextState();
+        // reveal state
+        await mineToNextState();
+        // propose state
+        const stakerIdAcc3 = await stakeManager.stakerIds(signers[3].address);
+        const staker = await stakeManager.getStaker(stakerIdAcc3);
+        const { biggestInfluencerId } = await getBiggestInfluenceAndId(stakeManager);
+
+        const iteration = await getIteration(voteManager, stakeManager, staker);
+        const tx = blockManager.connect(signers[3]).propose(epoch,
+          [],
+          iteration,
+          biggestInfluencerId);
+        assertRevert(tx, 'Cannot propose without revealing');
+      });
+      it('No Finalise Dispute should happen if no block is proposed or no one votes', async function () {
+        const epoch = await getEpoch();
+        await mineToNextState();
+        // dispute state
+        const sortedVotes = [];
+        const tx1 = blockManager.connect(signers[3]).giveSorted(epoch, 1, sortedVotes);
+        // const firstProposedBlockIndex = await blockManager.proposedBlocks(epoch, 0);
+        // const tx2 = blockManager.connect(signers[19]).finalizeDispute(epoch, firstProposedBlockIndex);
+        assert(tx1, 'should be able to give sorted votes');
+        // assertRevert(tx2, 'Error: Transaction reverted without a reason string');
```

Comment:
hey 
can we revert finalize dispute here as well

Reply:
@abhishekvispute hey, I did that before, but assertRevert is not catching the error, since there is no reason string returned. I even tried with the Error that came on revert but still it wasn't able to catch that. so I had to comment it.

Reply:
hmm don't give reason then 
only assetRevert

Reply:
I tried that also assertRevert(tx,'')

Reply:
hmm, no don't pass reason only, check if only assertRevert(tx) is working

Check this code 
https://github.com/razor-network/contracts/blob/0f54d6b86c36c0efb30f5682843f74008533bdd8/test/helpers/testHelpers.js#L76

see if there is no reason, it not checked, so mostly it should work

Reply:
ok